commands/operator-sdk/cmd/test/cluster.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2018 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmdtest
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/spf13/cobra"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+var (
+	testNamespace     string
+	kubeconfigCluster string
+	imagePullPolicy   bool
+)
+
+func NewTestClusterCmd() *cobra.Command {
+	testCmd := &cobra.Command{
+		Use:   "cluster <image name> [flags]",
+		Short: "Run End-To-End tests using image with embedded test binary",
+		RunE:  testClusterFunc,
+	}
+	defaultKubeConfig := ""
+	homedir, ok := os.LookupEnv("HOME")
+	if ok {
+		defaultKubeConfig = homedir + "/.kube/config"
+	}
+	testCmd.Flags().StringVarP(&testNamespace, "namespace", "n", "default", "Namespace to run tests in")
+	testCmd.Flags().StringVarP(&kubeconfigCluster, "kubeconfig", "k", defaultKubeConfig, "Kubeconfig path")
+	testCmd.Flags().BoolVarP(&imagePullPolicy, "imagePullPolicy", "i", false, "Set test pod image pull policy to 'Never'")
```

Comment:
k8s has three possible values for `imagePullPolicy`: `Always`, `Never`, and `IfNotPresent`. Even if you only expose two of those behaviors here, I don't think it's intuitive to map those values onto a boolean named `imagePullPolicy`. `true` and `false` are not policies.

What would you think of accepting `--pull=always` or `--pull=never`, with the default being `always`?

Or perhaps just expose a flag `--never-pull`?

I think one of those approaches, or something similar, would be more descriptive and understandable to the end user.

Reply:
Agreed. We should restrict the imagePullPolicy flag to Always/Never values so it's more obvious.